sections/nix_shell.zsh

```diff
@@ -0,0 +1,41 @@
+#
+# Nix Shell
+#
+# Nix can be used to provide some kind of virtual environment through the nix-shell command.
+# Link: https://nixos.org/manual/nix/stable/command-ref/nix-shell.html
+
+# ------------------------------------------------------------------------------
+# Configuration
+# ------------------------------------------------------------------------------
+
+SPACESHIP_NIXSHELL_SHOW="${SPACESHIP_NIXSHELL_SHOW=true}"
+SPACESHIP_NIXSHELL_ASYNC="${SPACESHIP_NIXSHELL_ASYNC=true}"
```

Comment:
I this should be `false` since this section only reads values from env variables and doesn't do any heavy computations.